pkg/k8s/watchers/node.go

```diff
@@ -154,49 +146,12 @@ func (u *ciliumNodeUpdater) OnDeleteNode(*v1.Node, *lock.StoppableWaitGroup) err
 	return nil
 }
 
-func (u *ciliumNodeUpdater) updateCiliumNode(kvStoreNodeUpdater KVStoreNodeUpdater, node *v1.Node) {
-	var (
-		controllerName = fmt.Sprintf("sync-node-with-ciliumnode (%v)", node.Name)
-
-		nodeSlim      = k8s.ConvertToNode(node.DeepCopy()).(*slim_corev1.Node)
-		k8sNodeParsed = k8s.ParseNode(nodeSlim, source.Local)
-	)
-
-	k8sNodeParsed.NodeIdentity = uint32(identity.ReservedIdentityHost)
-
-	doFunc := func(ctx context.Context) (err error) {
-		if option.Config.KVStore != "" && !option.Config.JoinCluster {
-			return kvStoreNodeUpdater.UpdateKVNodeEntry(k8sNodeParsed)
-		} else {
-			u.k8sWatcher.ciliumNodeStoreMU.RLock()
-			defer u.k8sWatcher.ciliumNodeStoreMU.RUnlock()
-
-			if u.k8sWatcher.ciliumNodeStore == nil {
-				return errors.New("CiliumNode cache store not yet initialized")
-			}
-
-			ciliumNodeInterface, exists, err := u.k8sWatcher.ciliumNodeStore.GetByKey(node.Name)
-			if err != nil {
-				return fmt.Errorf("failed to get CiliumNode resource from cache store: %w", err)
-			}
-			if !exists {
-				return nil
-			}
-
-			ciliumNode := ciliumNodeInterface.(*ciliumv2.CiliumNode).DeepCopy()
-
-			ciliumNode.Labels = node.GetLabels()
-
-			if _, err = k8s.CiliumClient().CiliumV2().CiliumNodes().Update(ctx, ciliumNode, metav1.UpdateOptions{}); err != nil {
-				return fmt.Errorf("failed to update CiliumNode labels: %w", err)
-			}
-		}
-
-		return nil
+func (u *ciliumNodeUpdater) updateCiliumNode(node *v1.Node) {
+	if node.Name != nodeTypes.GetName() {
+		// The cilium node updater should only update the information relevant
+		// to itself. It should not update any of the other nodes.
+		panic(fmt.Sprintf("trying to update node %q while we should only run for %q", node.Name, nodeTypes.GetName()))
```

Comment:
I wonder whether we should just log an error here and ignore the update. Or is this really an unrecoverable error which would prevent Cilium from working correctly going forward?

Reply:
This is a developer's mistake and should never happen. This event handler is only intended for the node where the agent is running. I will change the log files to report an error as BUG and ignore the update as you suggested.